Makefile

```diff
@@ -0,0 +1,106 @@
+# These variables get inserted into ./build/commit.go
+BUILD_TIME=$(shell date)
+GIT_REVISION=$(shell git rev-parse --short HEAD)
+GIT_DIRTY=$(shell git diff-index --quiet HEAD -- || echo "✗-")
+
+ldflags= -X github.com/NebulousLabs/skynet-accounts/build.GitRevision=${GIT_DIRTY}${GIT_REVISION} \
+-X "github.com/NebulousLabs/skynet-accounts/build.BuildTime=${BUILD_TIME}"
+
+racevars= history_size=3 halt_on_error=1 atexit_sleep_ms=2000
+
+# all will build and install release binaries
+all: release
+
+# count says how many times to run the tests.
+count = 1
+# pkgs changes which packages the makefile calls operate on. run changes which
+# tests are run during testing.
+pkgs = ./ ./api ./build ./database ./user
+
+# integration-pkgs defines the packages which contain integration tests
+integration-pkgs = ./test
+
+# fmt calls go fmt on all packages.
+fmt:
+	gofmt -s -l -w $(pkgs)
+
+# vet calls go vet on all packages.
+# We don't check composite literals because we need to use unkeyed fields for
+# MongoDB's BSONs and that sets vet off.
+# NOTE: go vet requires packages to be built in order to obtain type info.
+vet:
+	go vet -composites=false $(pkgs)
+
+# markdown-spellcheck runs codespell on all markdown files that are not
+# vendored.
+markdown-spellcheck:
+	git ls-files "*.md" :\!:"vendor/**" | xargs codespell --check-filenames
+
+# lint runs golangci-lint (which includes golint, a spellcheck of the codebase,
+# and other linters), the custom analyzers, and also a markdown spellchecker.
+lint: markdown-spellcheck lint-analyze
+	golangci-lint run -c .golangci.yml
```

Comment:
This file is still missing right? I'd get linting working asap to avoid the issues we have in Sia right now. Where it's already so late that we are having troubles enabling it.

Reply:
Added.

I used Sia's file + exclusion for `composites` errors for `govet` because we need those composites in order to talk to Mongo.